server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/RestoreInProgressAllocationDecider.java

```diff
@@ -39,7 +39,7 @@ public Decision canAllocate(final ShardRouting shardRouting, final RoutingAlloca
             return allocation.decision(Decision.YES, NAME, "not an API-level restore");
         }
 
-        final RestoreInProgress restoresInProgress = allocation.custom(RestoreInProgress.TYPE);
+        final RestoreInProgress restoresInProgress = allocation.getClusterState().custom(RestoreInProgress.TYPE);
```

Comment:
Why this change?

Reply:
I removed it to make RoutingAllocation simpler and not to mirror a lot of cluster state calls